The ContextMenu class is not serializable

Hi, thank you for this awesome extension. Just a minor thing: the ContextMenu class is not serializable: the Logger logger is not serializable and the serialization fails:

Caused by: java.io.NotSerializableException: java.util.logging.Logger
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1184)

Perhaps the logger field should be static.

--- a/vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/ContextMenu.java
+++ b/vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/ContextMenu.java
@@ -5,7 +5,6 @@ import java.util.ArrayList;
 import java.util.EventListener;
 import java.util.EventObject;
 import java.util.List;
-import java.util.logging.Logger;
 
 import com.vaadin.addon.contextmenu.ContextMenu.ContextMenuOpenListener.ContextMenuOpenEvent;
 import com.vaadin.addon.contextmenu.client.ContextMenuClientRpc;
@@ -25,15 +24,11 @@ import com.vaadin.util.ReflectTools;
 @SuppressWarnings("serial")
 public class ContextMenu extends AbstractExtension implements Menu {
 
-	private Logger logger = Logger.getLogger("ContextMenu");
-
 	private AbstractMenu menu = new AbstractMenu();
 
 	private ContextClickListener contextClickListener = new ContextClickListener() {
 		@Override
 		public void contextClick(ContextClickEvent event) {
-			logger.info("Context click listener");
-
 			fireEvent(new ContextMenuOpenEvent(ContextMenu.this, event));
 
 			open(event.getClientX(), event.getClientY());
@@ -80,8 +75,6 @@ public class ContextMenu extends AbstractExtension implements Menu {
 
 	@Override
 	public void beforeClientResponse(boolean initial) {
-		logger.info("beforeClientResponse");
-
 		super.beforeClientResponse(initial);
 
 		// FIXME: think about where this is supposed to be

--- a/vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/ContextMenu.java
+++ b/vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/ContextMenu.java
@@ -1,5 +1,6 @@
 package com.vaadin.addon.contextmenu;
 
+import java.io.Serializable;
 import java.lang.reflect.Method;
 import java.util.ArrayList;
 import java.util.EventListener;
@@ -196,7 +197,7 @@ public class ContextMenu extends AbstractExtension implements Menu {
 
 	/**** End of delegates to AbstractMenu ****/
 
-	public interface ContextMenuOpenListener extends EventListener {
+	public interface ContextMenuOpenListener extends EventListener, Serializable {
 
 		public static final Method MENU_OPENED = ReflectTools.findMethod(
 				ContextMenuOpenListener.class, "onContextMenuOpen",

--- a/vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/GridContextMenu.java
+++ b/vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/GridContextMenu.java
@@ -1,5 +1,6 @@
 package com.vaadin.addon.contextmenu;
 
+import java.io.Serializable;
 import java.util.EventListener;
 
 import com.vaadin.addon.contextmenu.ContextMenu.ContextMenuOpenListener.ContextMenuOpenEvent;
@@ -47,7 +48,7 @@ public class GridContextMenu extends ContextMenu {
         addGridSectionContextMenuListener(Section.BODY, listener);
     }
 
-    public interface GridContextMenuOpenListener extends EventListener {
+    public interface GridContextMenuOpenListener extends EventListener, Serializable {
 
         public void onContextMenuOpen(GridContextMenuOpenEvent event);
 

--- a/vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/Menu.java
+++ b/vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/Menu.java
@@ -5,7 +5,7 @@ import java.util.List;
 
 import com.vaadin.server.Resource;
 
-public interface Menu {
+public interface Menu extends Serializable {
 
 	/**
 	 * This interface contains the layer for menu commands of the

--- a/vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/MenuItem.java
+++ b/vaadin-context-menu-addon/src/main/java/com/vaadin/addon/contextmenu/MenuItem.java
@@ -1,11 +1,12 @@
 package com.vaadin.addon.contextmenu;
 
+import java.io.Serializable;
 import java.util.List;
 
 import com.vaadin.addon.contextmenu.Menu.Command;
 import com.vaadin.server.Resource;
 
-public interface MenuItem {
+public interface MenuItem extends Serializable {
 
 	/**
 	 * Checks if the item has children (if it is a sub-menu).

